packages/admin/src/Charcoal/Admin/ServiceProvider/AdminServiceProvider.php

```diff
@@ -177,6 +181,29 @@ protected function registerAdminServices(Container $container)
             };
         }
 
+        /**
+         * @param Container $container Pimple DI Container.
+         * @return FactoryInterface
+         */
+        $container['admin/email/factory'] = function (Container $container): FactoryInterface {
+            return new GenericFactory([
+                'map' => [
+                    'email' => Email::class
+                ],
+                'base_class' => EmailInterface::class,
+                'default_class' => Email::class,
+                'arguments' => [[
+                    'logger'             => $container['logger'],
+                    'config'             => $container['email/config'],
+                    'view'               => $container['admin/view'],
+                    'template_factory'   => $container['template/factory'],
+                    'queue_item_factory' => $container['model/factory'],
+                    'log_factory'        => $container['model/factory'],
+                    'tracker'            => $container['email/tracker']
+                ]]
```

Comment:
Do we want to define a `email/dependencies` service in `EmailServiceProvider`?

`EmailServiceProvider`:

```php
// The model dependencies might be already set from elsewhere; defines it if not.
if (!isset($container['email/dependencies'])) {
    /**
     * @param Container $container A Pimple DI container.
     * @return array The model dependencies array.
     */
    $container['email/dependencies'] = function (Container $container) {
        return [
            'logger'             => $container['logger'],
            'config'             => $container['email/config'],
            'view'               => $container['email/view'],
            'template_factory'   => $container['template/factory'],
            'queue_item_factory' => $container['model/factory'],
            'log_factory'        => $container['model/factory'],
            'tracker'            => $container['email/tracker'],
        ];
    };
}
```

`AdminServiceProvider`:

```php
$container['admin/email/factory'] = function (Container $container): FactoryInterface {
    $dependencies = $container['email/dependencies'];
    $dependencies['view'] = $container['admin/view'];

    return new GenericFactory([
        'map' => [
            'email' => Email::class
        ],
        'base_class' => EmailInterface::class,
        'default_class' => Email::class,
        'arguments' => [ $dependencies ],
    ]);
};
```